File: WebApi_CapstoneProject/Document Sharing System/Models/DTOs/NotificationDto.cs
namespace DSS.Models.DTOs
{
    public class NotificationDto
    {
        public Guid Id { get; set; }
        public string EntityName { get; set; } = string.Empty;
        public Guid EntityId { get; set; }
        public string Content { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
        public bool IsRead { get; set; }
        public DateTime? ReadAt { get; set; }
    }

    public class CreateNotificationDto
    {
        public string EntityName { get; set; } = string.Empty;
        public Guid EntityId { get; set; }
        public string Content { get; set; } = string.Empty;
        public List<Guid> UserIds { get; set; } = new List<Guid>();
    }

    public class MarkNotificationReadDto
    {
        public Guid NotificationId { get; set; }
    }
} 

File: Angular_Capstone/DSS/src/app/models/notificationDto.ts
export interface NotificationDto {
  id: string;
  entityName: string;
  entityId: string;
  content: string;
  createdAt: string;
  isRead: boolean;
  readAt: string | null;
}

export interface NotificationResponse {
  $id: string;
  success: boolean;
  data: {
    $id: string;
    $values: NotificationDto[];
  };
  error: string | null;
}

export interface UnreadCountDto {
  count: number;
}

export interface ApiResponseModel<T> {
  success: boolean;
  message: string;
  data?: T;
  errors?: string[];
} 